catalog/app/components/Preview/loaders/Json.js

```diff
@@ -58,16 +60,21 @@ function VegaLoader({ handle, children }) {
     },
     [signSpec, handle],
   )
-  return children(utils.useErrorHandling(processed, { handle, retry: data.fetch }))
+  const handled = utils.useErrorHandling(processed, { handle, retry: data.fetch })
+  const result =
+    gated && AsyncResult.Init.is(handled)
+      ? AsyncResult.Err(PreviewError.Gated({ handle, load: data.fetch }))
+      : handled
+  return children(result)
 }
 
 export const detect = R.either(utils.extIs('.json'), R.startsWith('.quilt/'))
 
 export const Loader = function JsonLoader({ handle, children }) {
   return utils.useFirstBytes({ bytes: 256, handle }).case({
     Ok: ({ firstBytes, contentLength }) =>
-      !!detectSchema(firstBytes) && contentLength <= MAX_SIZE ? (
-        <VegaLoader {...{ handle, children }} />
+      detectSchema(firstBytes) ? (
+        <VegaLoader {...{ handle, children, gated: contentLength > MAX_SIZE }} />
```

Comment:
What is the meaning of "gated"? If a file is too large, then we render it through a different "gate", or it does not fit into our gates?

Reply:
it's size-gated, meaning that it stays behind the gate if it's too large, waiting to be let in, idk smth like this lol

Reply:
Got it, thanks